embed/oko_attached/src/crypto/commit_reveal/session.ts

```diff
@@ -0,0 +1,327 @@
+/**
+ * Commit-Reveal Session Management
+ */
+
+import type { Bytes32 } from "@oko-wallet/bytes";
+import type { Result } from "@oko-wallet/stdlib-js";
+
+import type {
+  CommitRevealSession,
+  CommitRevealSessionState,
+  CreateSessionOptions,
+  NodeStatus,
+  EncryptedToken,
+} from "./types";
+import type { ClientEcdheKeypair } from "./utils";
+import {
+  generateSessionId,
+  generateClientKeypair,
+  calculateTokenHash,
+  calculateExpiresAt,
+  isSessionExpired,
+} from "./utils";
+
+// ============================================================================
+// Session Creation
+// ============================================================================
+
+export interface CreateSessionResult {
+  session: CommitRevealSession;
+  keypair: ClientEcdheKeypair;
+}
+
+/**
+ * Create a new commit-reveal session.
+ */
+export function createSession(
+  options: CreateSessionOptions,
+): Result<CreateSessionResult, string> {
+  const sessionId = generateSessionId();
+
+  const keypairResult = generateClientKeypair();
+  if (!keypairResult.success) {
+    return { success: false, err: keypairResult.err };
+  }
+
+  const tokenHashResult = calculateTokenHash(
+    options.oauth_token,
+    options.sdk_version,
+  );
+  if (!tokenHashResult.success) {
+    return { success: false, err: tokenHashResult.err };
+  }
+
+  const now = new Date();
+
+  const session: CommitRevealSession = {
+    session_id: sessionId,
+    session_type: "OAUTH_COMMIT_REVEAL",
+    client_public_key: keypairResult.data.publicKey,
+    sdk_version: options.sdk_version,
+    user_email: options.user_email,
+    public_key: options.wallet_public_key,
+    token_hash: tokenHashResult.data,
+    state: "INITIALIZED",
+    created_at: now,
+    updated_at: now,
+    expires_at: calculateExpiresAt(now),
+    commit_phase: {
+      nodes_committed: [],
+      total_nodes: options.node_urls.length,
+      encrypted_tokens: {},
+      node_public_keys: {},
+    },
+    reveal_phase: {
+      nodes_revealed: [],
+      total_nodes: options.node_urls.length,
+    },
+    operation_type: options.operation_type,
+  };
+
+  return {
+    success: true,
+    data: { session, keypair: keypairResult.data },
+  };
+}
+
+// ============================================================================
+// State Management
+// ============================================================================
+
+/**
+ * Update session state.
+ */
+export function updateState(
+  session: CommitRevealSession,
+  newState: CommitRevealSessionState,
+): CommitRevealSession {
+  return {
+    ...session,
+    state: newState,
+    updated_at: new Date(),
+  };
+}
+
+/**
+ * Check if state transition is valid.
+ */
+export function canTransitionTo(
+  session: CommitRevealSession,
+  targetState: CommitRevealSessionState,
+): boolean {
+  if (isSessionExpired(session.expires_at)) {
+    return targetState === "TIMEOUT";
+  }
+
+  const transitions: Record<
```

Comment:
I rather find this to be a little more complicated than it should be. But let's just go with it for now